azure/modules/aks/variables.tf

```
@@ -0,0 +1,107 @@
+# Copyright 2021 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Common variables.
+variable "location" {
+  description = "Azure location where all the resources being created."
+  type        = string
+}
+
+variable "name" {
+  description = "Name is the prefix to use for resources that needs to be created."
+  type        = string
+}
+
+variable "environment" {
+  description = "Name of the environment where infrastructure being built."
+  type        = string
+}
+
+# Kubernetes cluster configuration variables
+variable "kubernetes_version" {
+  description = "Version of the AKS cluster."
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group in which the resources will be created."
+  type        = string
+}
+
+variable "user_assigned_id" {
+  description = "The name of the user identity in which the resources will be created."
+  type        = string
+}
+
+# Variables for the default node_pool.
+variable "private_subnet" {
+  description = "The subnet id of the virtual network where the virtual machines will reside."
+  type        = string
+}
+
+variable "vm_size" {
+  description = "Specifies the size of the virtual machine."
+  default     = "Standard_DS2_v2"
```

This instance type is too small.

```suggestion
  default     = "Standard_E8_v4"
```